src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/RazorHoverEndpoint.cs

```diff
@@ -51,22 +54,35 @@ internal class RazorHoverEndpoint : AbstractRazorDelegatingEndpoint<VSHoverParam
         protected override string EndpointName => LanguageServerConstants.RazorHoverEndpointName;
 
         /// <inheritdoc/>
-        protected override DelegatedHoverParams CreateDelegatedParams(VSHoverParamsBridge request, DocumentContext documentContext, Projection projection)
-            => new DelegatedHoverParams(
-                documentContext.Identifier,
-                projection.Position,
-                projection.LanguageKind);
+        protected override async Task<DelegatedHoverParams> CreateDelegatedParamsAsync(VSHoverParamsBridge request, CancellationToken cancellationToken)
+        {
+            var documentContext = await _documentContextFactory.GetRequiredAsync(request.TextDocument.Uri, cancellationToken).ConfigureAwait(false);
```

Comment:
This could be left for me to do, in the next PR, but I wonder how people feel about an abstract `GetTestDocumentIndentifier` method, so that the base can get the document context?

This is the same pattern I did in Roslyn, and so far is ending up in CLaSP, so might be okay since its consistent?

Reply:
Given it'd be consistent with CLaSP i'd be OK with it. Would probably also set us up for success in that model. I could also imagine it being a `GetDocumentContextAsync` too where it side-stepped the identifier entirely

Reply:
The downside to that is that it would require inheritors to store the document context factory, which they may otherwise never use, which is a little messy to me, but not a huge deal.